test_cluster_detection.py

```diff
@@ -0,0 +1,242 @@
+        
+        # Test
+        assert(not model_ran)
+        assert(detections_in_world is None)
+    
+    def test_at_min_total_threshold(self, cluster_model:ClusterEstimation):
+        """
+        Should run once once total threshold reached regardless of
+        current bucket size
+        """
+        # Setup
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1  # should not run the first time
+        NEW_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD - 1 # under 10 new points 
+        
+        generated_detections, labels, cluster_positions = generate_cluster_data([NUM_DATA_POINTS])
+        generated_detections_2, labels_2, cluster_positions_2 = generate_cluster_data([NEW_DATA_POINTS])
+
+        # Run
+        cluster_model.clear_all_data()
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+        model_ran, detections_in_world = cluster_model.run(generated_detections_2, False)
+
+        # Test
+        assert(model_ran)
+        assert(detections_in_world is not None)
+
+    def test_under_min_bucket_size(self, cluster_model:ClusterEstimation):
+        """
+        New data under threshold should not run 
+        """
+        # Setup
+        NUM_DATA_POINTS = MIN_TOTAL_POINTS_THRESHOLD + 10  # should run the first time
+        NEW_DATA_POINTS = MIN_NEW_POINTS_THRESHOLD - 1 # under 10 new points 
+
+        generated_detections, labels, cluster_positions= generate_cluster_data([NUM_DATA_POINTS])
+        generated_detections_2, labels_2, cluster_positions_2 = generate_cluster_data([NEW_DATA_POINTS])
+
+        # Run
+        cluster_model.clear_all_data()
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+        model_ran, detections_in_world = cluster_model.run(generated_detections_2, False)
+
+        # Test
+        assert(not model_ran)
+        assert(detections_in_world is None)
+
+    def test_good_data(self, cluster_model:ClusterEstimation):
+        """
+        All conditions met should run
+        """
+        EXPECTED_CLUSTER_COUNT = MIN_TOTAL_POINTS_THRESHOLD + 1  # more than min total threshold should run 
+        generated_detections, labels, cluster_positions= generate_cluster_data([EXPECTED_CLUSTER_COUNT])
+    
+        # Run
+        cluster_model.clear_all_data()
+        model_ran, detections_in_world = cluster_model.run(generated_detections, False)
+        
+        # Test
+        assert(model_ran)
+        assert(detections_in_world is not None)
+    
+
+class TestCorrectNumberClusterOutputs():
+    """
+    Tests if cluster estimation output matches number of input clusters
+    at different data scenarios. 
+    """
+    def test_detect_normal_data(self, cluster_model:ClusterEstimation):
+        """
+        data with small distribution and equal number of points per cluster center. 
+        """
+        # Setup
+        STD_DEV_REGULAR = CENTER_BOX_SIZE / 500
```

Comment:
If `CENTER_BOX_SIZE = 5000` , would this still be valid?

Reply:
Good point, I'm assuming the CENTER_BOX_SIZE as the theoretical max "world-size", i.e. the maximum area were we will have detections. STD_DEV_REG is now  = 1, meaning a 1m standard deviation detections is assumed for the tests. This can be tuned in the future depending on how the detection goes.